packages/strapi-admin/admin/src/components/LeftMenuLinkContainer/index.js

```diff
@@ -40,109 +37,70 @@ function LeftMenuLinkContainer({ plugins, ...rest }) {
+  const pluginsLinks = Object.values(plugins)
+    .filter(
+      plugin =>
         plugin.id !== 'email' &&
         plugin.id !== 'content-manager' &&
-        shouldInjectPlugin
-      ) {
-        const pluginSuffixUrl = plugin.suffixUrl
-          ? plugin.suffixUrl(plugins)
-          : '';
+        !!plugin.mainComponent
+    )
+    .map(plugin => {
+      const pluginSuffixUrl = plugin.suffixUrl ? plugin.suffixUrl(plugins) : '';
 
-        const destination = `/plugins/${get(plugin, 'id')}${pluginSuffixUrl}`;
-
-        return (
-          <LeftMenuLink
-            {...rest}
-            key={get(plugin, 'id')}
-            icon={get(plugin, 'icon') || 'plug'}
-            label={get(plugin, 'name')}
-            destination={destination}
-            pluginSuffixUrl={pluginSuffixUrl}
-            suffixUrlToReplaceForLeftMenuHighlight={
-              plugin.suffixUrlToReplaceForLeftMenuHighlight || ''
-            }
-          />
-        );
-      }
-    })
-  ) : (
-    <li key="emptyList" className="noPluginsInstalled">
-      <FormattedMessage {...messages.noPluginsInstalled} key="noPlugins" />.
-    </li>
-  );
+      return {
+        icon: get(plugin, 'icon') || 'plug',
+        label: get(plugin, 'name'),
+        destination: `/plugins/${get(plugin, 'id')}${pluginSuffixUrl}`,
+      };
+    });
 
-  const staticLinks = [
-    {
-      icon: 'list',
-      label: messages.listPlugins.id,
-      destination: '/list-plugins',
-    },
-    {
-      icon: 'shopping-basket',
-      label: messages.installNewPlugin.id,
-      destination: '/marketplace',
+  const menu = {
+    ...contentTypesSections,
+    plugins: {
+      searchable: false,
+      name: 'plugins',
+      emptyLinksListMessage: messages.noPluginsInstalled.id,
+      links: pluginsLinks,
     },
-    {
-      icon: 'cog',
-      label: messages.settings.id,
-      destination: SETTINGS_BASE_URL,
+    general: {
+      searchable: false,
+      name: 'general',
+      links: [
+        {
+          icon: 'list',
+          label: messages.listPlugins.id,
+          destination: '/list-plugins',
+        },
+        {
+          icon: 'shopping-basket',
+          label: messages.installNewPlugin.id,
+          destination: '/marketplace',
+        },
+        {
+          icon: 'cog',
+          label: messages.settings.id,
+          destination: SETTINGS_BASE_URL,
+        },
+      ],
     },
-  ];
+  };
 
   return (
     <Wrapper>
-      {linkSections}
-      <div>
-        <p className="title">
-          <FormattedMessage {...messages.plugins} />
-        </p>
-        <ul className="list">{pluginsLinks}</ul>
-      </div>
-      <div>
-        <p className="title">
-          <FormattedMessage {...messages.general} />
-        </p>
-        <ul className="list">
-          {staticLinks.map(link => (
-            <LeftMenuLink {...rest} key={link.destination} {...link} />
-          ))}
-        </ul>
-      </div>
+      {Object.keys(menu).map(current => (
```

Comment:
The `menu` object should be a single source of truth for the left menu now.

Reply:
Great!